contracts/colonyNetwork/ColonyNetworkStorage.sol

```diff
@@ -127,15 +148,23 @@ contract ColonyNetworkStorage is ColonyNetworkDataTypes, DSMath, CommonStorage {
   // All colonies are able to manage their Local (domain associated) skills
   modifier allowedToAddSkill(bool globalSkill) {
     if (globalSkill) {
+      assert(isMiningChain());
       require(msgSender() == metaColony, "colony-must-be-meta-colony");
     } else {
       require(_isColony[msgSender()] || msgSender() == address(this), "colony-caller-must-be-colony");
     }
     _;
   }
 
-  modifier skillExists(uint skillId) {
+  modifier skillExists(uint256 skillId) {
     require(skillCount >= skillId, "colony-invalid-skill-id");
+    require(isMiningChain() || (skillId >> 128) == getChainId() , "colony-invalid-skill-id");
```

Comment:
Could use `toChainId` here, which would entail moving the function down into the "internal" section